Geolocator/Geolocator/Geolocator.Plugin.iOS/GeolocatorImplementation.cs

```diff
@@ -207,38 +208,45 @@ public Task<bool> StartListeningAsync(int minTime, double minDistance, bool incl
+
+			// keep reference to settings so that we can stop the listener appropriately later
+			listenerSettings = settings;
+
             double desiredAccuracy = DesiredAccuracy;
 
-			// if we have listener settings, apply them to the locator.
-			if (settings != null)
-			{
-				// set background flag
-				if (UIDevice.CurrentDevice.CheckSystemVersion(9, 0))
-					manager.AllowsBackgroundLocationUpdates = settings.AllowBackgroundUpdates;
+			#region apply settings to location manager
+			// set background flag
+			if (UIDevice.CurrentDevice.CheckSystemVersion(9, 0))
+				manager.AllowsBackgroundLocationUpdates = settings.AllowBackgroundUpdates;
 
-				// configure location update pausing
-				if (UIDevice.CurrentDevice.CheckSystemVersion(6, 0))
-				{
-					manager.PausesLocationUpdatesAutomatically = settings.PauseLocationUpdatesAutomatically;
-
-					if (settings.ActivityType == ActivityType.AutomotiveNavigation)
-						manager.ActivityType = CLActivityType.AutomotiveNavigation;
-					else if (settings.ActivityType == ActivityType.Fitness)
-						manager.ActivityType = CLActivityType.Fitness;
-					else if (settings.ActivityType == ActivityType.Other)
-						manager.ActivityType = CLActivityType.Other;
-					else if (settings.ActivityType == ActivityType.OtherNavigation)
-						manager.ActivityType = CLActivityType.OtherNavigation;
-				}
+			// configure location update pausing
+			if (UIDevice.CurrentDevice.CheckSystemVersion(6, 0)) 
+			{
+				manager.PausesLocationUpdatesAutomatically = settings.PauseLocationUpdatesAutomatically;
+
+				if (settings.ActivityType == ActivityType.AutomotiveNavigation)
+					manager.ActivityType = CLActivityType.AutomotiveNavigation;
+				else if (settings.ActivityType == ActivityType.Fitness)
+					manager.ActivityType = CLActivityType.Fitness;
+				else if (settings.ActivityType == ActivityType.Other)
+					manager.ActivityType = CLActivityType.Other;
+				else if (settings.ActivityType == ActivityType.OtherNavigation)
+					manager.ActivityType = CLActivityType.OtherNavigation;
+			}
 
-				// to use deferral, CLLocationManager.DistanceFilter must be set to CLLocationDistance.None, and CLLocationManager.DesiredAccuracy must be 
-				// either CLLocation.AccuracyBest or CLLocation.AccuracyBestForNavigation. deferral only available on iOS 6.0 and above.
-				if (CanDeferLocationUpdate && settings.DeferLocationUpdates)
-				{
-					minDistance = CLLocationDistance.FilterNone;
-					desiredAccuracy = CLLocation.AccuracyBest;
-				}
+			// to use deferral, CLLocationManager.DistanceFilter must be set to CLLocationDistance.None, and CLLocationManager.DesiredAccuracy must be 
+			// either CLLocation.AccuracyBest or CLLocation.AccuracyBestForNavigation. deferral only available on iOS 6.0 and above.
+			if (CanDeferLocationUpdate && settings.DeferLocationUpdates)
+			{
+				minDistance = CLLocationDistance.FilterNone;
+				desiredAccuracy = CLLocation.AccuracyBest;
 			}
+			#endregion
```

Comment:
The only change above is that I removed the `settings != null` check since we should instantiate the default settings if none are passed in.